### ISSUE TITLE
useCheckbox does not prevent form submittion is isRequired === true

### ISSUE THREAD

**Issue by mJarsater** [adobe/react-spectrum#7317]
### Provide a general summary of the issue here

When using the useCheckbox hook to contstruct a custom checkbox the isRequired prop does not prevent form submittions.

### 🤔 Expected Behavior?

The isRequired prop should work the same way when using the Checkbox component vs the useCheckbox hook.

### 😯 Current Behavior

It seems like it is getting the correct aria attribute: `aria-required="true"` but it does not prevent form sunmittions.

### 🔦 Context

```json
    "react-aria": "3.35.0",
    "react-aria-components": "1.4.0",
     "react-stately": "3.33.0",
```

### 🖥️ Steps to Reproduce

https://codesandbox.io/p/sandbox/fc6w74

### Version

1.4.1

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What operating system are you using?

macOS 15.0.1 (24A348)

**Comment by nwidynski** [comment]
@mJarsater Thanks for reporting this 👍 

Afaik, React Aria Components do not aim to provide the same defaults as the hooks. You can achieve the same behavior with the hook by setting the `validationBehavior` property to `native`. This is done by default inside the `<Checkbox />`:

https://github.com/adobe/react-spectrum/blob/5ed06068ee2742f32e066ffa8eb55fd93a083123/packages/react-aria-components/src/Checkbox.tsx#L177C2-L177C91

**Comment by devongovett** [comment]
Yes, that's correct. The hooks have `validationBehavior="aria"` by default, because that was the original behavior before we added native validation and changing the default would be a breaking change.
